src/ast/passes/codegen_llvm.cpp

```diff
@@ -2281,6 +2281,19 @@ void CodegenLLVM::visit(Probe &probe)
         matches = bpftrace_.probe_matcher_->get_matches_for_ap(*attach_point);
       }
 
+      probe_count_ += matches.size();
+      if (probe_count_ > bpftrace_.max_programs_)
+      {
+        throw std::runtime_error(
```

Comment:
Using `LOG(FATAL)` here might be nicer for the end user

Reply:
I wouldn't say, the exception is caught and translated to `LOG(ERROR)` while `LOG(FATAL)` will terminate the program with `SIGABRT` which is not very user-friendly.

Reply:
ah we catch it, nice :)